2377/2/versus/versus.rb

```diff
@@ -0,0 +1,164 @@
+# rubocop:disable Metrics/LineLength
+# rubocop:disable Metrics/MethodLength
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/AbcSize
+# rubocop:disable Style/GuardClause
+
+option_bad_words = ARGV[0].to_s[/(?<=--top-bad-words=)\d*/]
+option_words = ARGV[0].to_s[/(?<=--top-words=)\d*/]
+ARGV.each { |arg| @option_name = arg.to_s[/(?<=--name=)(\w|\d|.|\s)*/] }
+
+@general_bad_words = [/сук([аи])?[м]?[и]?/, /бля[тд]?[ьие]?[й]?/, /(объ)?(за)?[её]ба[тнл]?[ьыаиос]?[йя]?/, /(от)?пизд[еиаы][тцшб]?[оь]?[л]?[ыи]?(ть)?/, /(о)?(на)?ху[йеия][тлвн]?[ьаиешн]?[ыеиола]?[яйгьм]?[оун]?[ыа]?[йяхе]?/, /(на)?хер[а]?/, /пидор[ауоеы]?[св]?[ыо]?(в)?/, /мраз[ьи]/, /еблан[ыоа]?[мв]?[и]?/, /ублюд[ок][киа]?[м]?/, /хуесос[ыао]?[мв]?/, /гандон[ыа]?/, /жоп[аы]/, /у[ёе]б[каоы]?[ки]?/, /\W*\*\W*/]
+@words = Hash.new 0
+@battles = Hash.new 0
+@words_per_battle = Hash.new 0
+@rounds = Hash.new { |h, k| h[k] = [0, 0] }
+@words_per_round = Hash.new 0
```

Comment:
what about creating class for rapper which will store all these data for only one of them?